src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -3513,35 +3514,33 @@ bool SvgLoader::header()
         viewFlag = loaderData.doc->node.doc.viewFlag;
         align = loaderData.doc->node.doc.align;
         meetOrSlice = loaderData.doc->node.doc.meetOrSlice;
-        w = 1.0f;
-        h = 1.0f;
 
-        //Return the brief resource info such as viewbox:
-        if (viewFlag & SvgViewFlag::Width) {
-            w = loaderData.doc->node.doc.w;
-        }
-        if (viewFlag & SvgViewFlag::Height) {
-            h = loaderData.doc->node.doc.h;
-        }
-        //Override size
-        if (viewFlag & SvgViewFlag::Viewbox) {
-            vx = loaderData.doc->node.doc.vx;
-            vy = loaderData.doc->node.doc.vy;
-            vw = loaderData.doc->node.doc.vw;
-            vh = loaderData.doc->node.doc.vh;
+        vx = loaderData.doc->node.doc.vx;
```

Comment:
loaderData.doc->node.doc.vx, vy is probably not initialized.

Reply:
For simplication, let it expect the proper values are returned. Otherwise it will override them after the sequence. (3529 ~ 3535)

Reply:
In your code (532~3535), only vw and vh are overried. If `<viewbox>` is not defined, no values are set for loaderData.doc->node.doc.vx and loaderData.doc->node.doc.vy.
(The old code is logically set only when vx and vy are initialized in the class and the viewbox is declared.)
It's common for structs to be set to 0, but I know not all compilers guarantee this.
Therefore, in _createSvgNode(), I think we need to explicitly initialize the values of `doc`


`vx, vy -999.000000 -999.000000 loaderData.doc->node.doc vx vy -999.000000 -999.000000`

```diff
diff --git a/src/loaders/svg/tvgSvgLoader.cpp b/src/loaders/svg/tvgSvgLoader.cpp
index 03a8fee1..36f33ee9 100644
--- a/src/loaders/svg/tvgSvgLoader.cpp
+++ b/src/loaders/svg/tvgSvgLoader.cpp
@@ -1347,6 +1347,10 @@ static SvgNode* _createSvgNode(SvgLoaderData* loader, SvgNode* parent, const cha
     doc->align = AspectRatioAlign::XMidYMid;
     doc->meetOrSlice = AspectRatioMeetOrSlice::Meet;
     doc->viewFlag = SvgViewFlag::None;
+
+    doc->vx = -999; // Cases where the compiler does not guarantee initialization
+    doc->vy = -999;
+
     func(buf, bufLength, _attrParseSvgNode, loader);

     if (!(doc->viewFlag & SvgViewFlag::Viewbox)) {
@@ -3535,6 +3539,7 @@ bool SvgLoader::header()
             else vh = h = 1.0f;
             run(0);
         }
+        printf("vx, vy %f %f loaderData.doc->node.doc vx vy %f %f\n", vx, vy, loaderData.doc->node.doc.vx,loaderData.doc->node.doc.vy);

         return true;
     }

```

```diff
--- a/src/examples/images/favorite_on.svg
+++ b/src/examples/images/favorite_on.svg
@@ -2,7 +2,7 @@
 <!-- Generator: Adobe Illustrator 16.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
 <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
 <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
-        width="80px" height="80px" viewBox="-14 -14 80 80" enable-background="new -14 -14 80 80" xml:space="preserve">
+        width="80px" height="80px"  enable-background="new -14 -14 80 80" xml:space="preserve">
```

Reply:
overriding code has been missed by my fault. now it does.